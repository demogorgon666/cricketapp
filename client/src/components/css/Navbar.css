@import url("https://fonts.googleapis.com/css?family=Montserrat:400,500,600&display=swap");
body {
	margin: 0px;
	letter-spacing: 0.3px;
}
.nav-parent {
	background-color: #2980b9;
	position: fixed;
	top: 0;
	left: 0;
	right: 0;
	box-shadow: 0px 1px 2px 0px #ececec;
	z-index: 3;
}
nav {
	padding: 5px 20px;
}
#nav-brand {
	color: #ffffff;
	text-decoration: none;
	font-size: 16px;
}
.menu .link {
	color: #ffffff;
	text-decoration: none;
	display: block;
}
.menu li {
	font-size: 16px;
	padding: 10px 0px;
}
.genders {
	display: flex;
	flex-direction: row;
	border: 1px solid #ffffff;
	border-radius: 4px;
}
.genders > span {
	padding: 8px;
	font-size: 10px;
	color: #ffffff;
}
.genders > span:not(:first-child) {
	border-left: 1px solid #ffffff;
}
.genders > span:hover {
	cursor: pointer;
	transition: 0.3s ease;
	background-color: #ffffff;
	color: #272727;
}
.genders > span:hover:last-child {
	border-left: 1px solid #272727;
}
.genders > span:hover:first-child {
	border-right: 1px solid #272727;
}

.genders > span.active-gender:hover {
	border-left: 0px;
	border-right: 0px;
}

/* Mobile menu */
.menu ul {
	margin: 0px;
	padding: 0px;
	list-style: none;
	display: flex;
	flex-wrap: wrap;
	justify-content: space-between;
	align-items: center;
}
.genders {
	order: 1;
}

.menu-icon {
	cursor: pointer;
	order: 2;
}

.item {
	text-align: center;
	order: 3;
	width: 100%;
	display: none;
}
.search-box-li {
	order: 6;
	width: 100%;
}
.search-icon {
	order: 4;
}
.login-icon {
	order: 5;
	width: 100%;
	text-align: center;
}
.search-box {
	margin: 0px;
	padding: 10px;
	font-size: 12px;
	border: 0px solid #707070;
	border-radius: 4px;
}
.search-icon {
	display: none;
}
.menu-btn {
	display: none;
}

.bars {
	background-color: #ffffff;
	height: 2px;
	width: 20px;
	display: inline-block;
	position: relative;
}

.bars::before,
.bars::after {
	background-color: #ffffff;
	height: 2px;
	width: 20px;
	display: inline-block;
	position: absolute;
	content: "";
}

.bars::before {
	top: -10px;
}

.bars::after {
	top: -5px;
}

.fa-search {
	color: #ffffff;
}
.search-result {
	position: absolute;
	background-color: #ffffff;
	border-radius: 4px;
	height: auto;
	margin-top: 2px;
	box-shadow: 0px 0px 2px 0px #c2c2c2;
}
.search-result > div {
	padding: 6px;
}
.search-result > div:hover {
	background-color: #e6e6e6;
	cursor: pointer;
}

.search-result > div:first-child {
	border-radius: 4px 4px 0px 0px;
}
.search-result > div:last-child {
	border-radius: 0px 0px 4px 4px;
}
.search-result > .search-no-result:hover {
	border-radius: 4px;
	background-color: #ffffff;
	cursor: default;
}
.search-list-item {
	font-weight: 500;
	color: #272727;
	text-decoration: none;
}

/* admin properties */

.menu .admin-ul {
	justify-content: end;
}

.admin-nav-brand {
	margin-right: auto;
}

@media screen and (min-width: 526px) {
	.item,
	.login-icon {
		width: auto;
	}
	.item .link {
		font-size: 12px;
	}
	#adminteam {
		margin-left: 30px;
	}
}

@media screen and (min-width: 768px) {
	#nav-brand {
		font-size: 20px;
	}
	.item .link {
		font-size: 14px;
	}
	.search-box-li {
		width: auto;
		order: 4;
	}
	.nav-parent {
		padding: 0px 40px;
	}
}

@media screen and (min-width: 992px) {
	.item .link {
		font-size: 16px;
	}
	.genders > span {
		font-size: 14px;
	}
	.nav-brand-li {
		/* flex-grow: 3; */
		min-width: 30vw;
	}
	.menu.admin {
		padding: 5px 130px;
	}
}
.active-gender {
	background-color: #ffffff;
	color: #272727 !important;
}
