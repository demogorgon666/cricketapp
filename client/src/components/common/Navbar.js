import React, { Component } from "react";
import "../css/Navbar.css";
import { Link, Redirect } from "react-router-dom";
import { getSearch } from "../../actions/SerachAction";
import { connect } from "react-redux";
import decode from "jwt-decode";

let decoded_token;

export class Navbar extends Component {
	state = {
		active: true,
		searchInput: "",
		isChecked: window.innerWidth >= 526 ? true : false,
		width: window.innerWidth,
		redirect: false,
		pageLink: this.props.pageLink ? this.props.pageLink : "",
		showGender: this.props.showGender
	};

	toggleChange = pageLink => {
		this.setState({ pageLink });
		window.innerWidth <= 526
			? this.setState({ isChecked: !this.state.isChecked })
			: console.log("do nothing");
	};

	handleSearchInputChange = e => {
		this.setState({ [e.target.name]: e.target.value });
		let search_term = {
			search: this.state.searchInput
		};
		this.props.getSearch(search_term, this.props.history);
	};

	componentDidMount() {
		window.addEventListener("resize", this.updateDimensions);
		if (localStorage.getItem("token")) {
			decoded_token = decode(localStorage.getItem("token"));
		}
	}

	updateDimensions = () => {
		this.setState({
			width: window.innerWidth,
			isChecked: window.innerWidth >= 526 ? true : false
		});
	};

	renderRedirect = () => {
		if (this.state.redirect) {
			return <Redirect to="/login" />;
		}
	};

	logout = () => {
		localStorage.removeItem("token");
		this.setState({ redirect: true });
	};

	render() {
		return (
			<div className="nav-parent">
				{this.renderRedirect()}
				<nav className="menu">
					<ul>
						{/* brand name which is at left of navbar */}
						<li className="nav-brand-li">
							<Link
								to="/"
								id="nav-brand"
								onClick={() => {
									this.setState({ pageLink: "" });
								}}
							>
								CricketAlpha
							</Link>
						</li>

						{/* toggle gender buttons for men and women */}
						<div
							className="genders"
							style={{
								display: this.state.showGender
									? "default"
									: "none"
							}}
						>
							<span
								id="men"
								className={
									this.props.gender === "male"
										? "active-gender"
										: ""
								}
								onClick={() => this.props.changeGender("male")}
							>
								Men
							</span>
							<span
								id="women"
								className={
									this.props.gender === "female"
										? "active-gender"
										: ""
								}
								onClick={() =>
									this.props.changeGender("female")
								}
							>
								Women
							</span>
						</div>

						{/* hidden checkbox to show and hide the menu in small screen device */}
						<input
							type="checkbox"
							id="menu-btn"
							className="menu-btn"
							checked={this.state.isChecked}
							onClick={() =>
								this.toggleChange(this.state.pageLink)
							}
						/>

						{/* label for checkbox which makes the hamburger menu */}
						<label
							className="menu-icon"
							for="menu-btn"
							style={{
								display:
									this.state.width < 526 ? "block" : "none"
							}}
						>
							<span className="bars"></span>
						</label>

						{/* matches link */}
						<li
							className={
								this.state.pageLink == "matches"
									? "item active-link"
									: "item"
							}
							style={{
								display: this.state.isChecked ? "block" : "none"
							}}
							onClick={() => this.toggleChange("matches")}
						>
							<Link className="link" to="/matches" id="matches">
								Matches
							</Link>
						</li>

						{/* teams link */}
						<li
							className={
								this.state.pageLink == "teams"
									? "item active-link"
									: "item"
							}
							style={{
								display: this.state.isChecked ? "block" : "none"
							}}
							onClick={() => this.toggleChange("teams")}
						>
							<Link className="link" to="/teams" id="teams">
								Teams
							</Link>
						</li>

						{/* players link */}
						<li
							className={
								this.state.pageLink == "players"
									? "item active-link"
									: "item"
							}
							style={{
								display: this.state.isChecked ? "block" : "none"
							}}
							onClick={() => this.toggleChange("players")}
						>
							<Link className="link" to="/players" id="players">
								Players
							</Link>
						</li>

						{/* search box list item */}
						<li className="search-box-li">
							{/* search box to search for players and teams */}
							<input
								id="searchInput"
								name="searchInput"
								className="search-box"
								value={this.state.searchInput}
								onChange={this.handleSearchInputChange}
								placeholder="Search for Team or Player"
							/>

							{/* shows search result after entering text in search boc */}

							<div
								className="search-result"
								style={{
									display: this.state.searchInput
										? "block"
										: "none"
								}}
							>
								{/* checks whether search has entries or not 
								if not then prints no result found */}
								{this.props.search.length !== 0 ? (
									this.props.search.player.length !== 0 ||
									this.props.search.team.length !== 0 ? (
										<>
											{/* checks for player array length in seaarch
										if present map all the player available */}
											{this.props.search.player.length !==
											0
												? this.props.search.player.map(
														mapped_search => (
															<div>
																<Link
																	className="search-list-item"
																	to={{
																		pathname:
																			"/playerInfo/" +
																			mapped_search.player_id
																		// state:{
																	}}
																>
																	<div
																		style={{
																			backgroundImage: `url(data:image/jpeg;base64,${mapped_search.player_image}`
																		}}
																	></div>
																	<div
																		className="search-suggestion"
																		id={
																			"player-search-" +
																			mapped_search.player_id
																		}
																	>
																		{
																			mapped_search.player_name
																		}
																	</div>
																</Link>
															</div>
														)
												  )
												: null}

											{/* checks for team array length in seaarch
										if present map all the team available */}
											{this.props.search.team.length !== 0
												? this.props.search.team.map(
														mapped_search => (
															<div>
																<Link
																	className="search-list-item"
																	to={{
																		pathname:
																			"/teamdetails/" +
																			mapped_search.team_id
																		// state:{
																	}}
																>
																	<div
																		style={{
																			backgroundImage: `url(data:image/jpeg;base64,${mapped_search.team_image}`
																		}}
																	></div>
																	<div
																		className="search-suggestion"
																		id={
																			"team-search-" +
																			mapped_search.team_id
																		}
																	>
																		{
																			mapped_search.team_name
																		}
																	</div>
																</Link>
															</div>
														)
												  )
												: null}
										</>
									) : (
										<div className="search-no-result">
											<div style={{ color: "#272727" }}>
												No result found
											</div>
										</div>
									)
								) : null}
							</div>
						</li>

						{/* icon for login */}
						<li
							className="login-icon"
							style={{
								display: this.state.isChecked ? "block" : "none"
							}}
						>
							<Link className="link" to="/login">
								<i className="fas fa-user"></i>
							</Link>
							<div className="user-options-links">
								{!localStorage.getItem("token") ? (
									<>
										<div>
											<Link
												className="user-options-item"
												to="/login"
											>
												<div id="login">Login</div>
											</Link>
										</div>
										<div>
											<Link
												className="user-options-item"
												to="/register"
											>
												<div id="register">
													Register
												</div>
											</Link>
										</div>
									</>
								) : decoded_token ? (
									decoded_token.isAdmin ? (
										<>
											<div>
												<Link
													className="user-options-item"
													to="/adminteam"
												>
													<div id="manage-teams">
														Manage Teams
													</div>
												</Link>
											</div>
											<div>
												<Link
													className="user-options-item"
													to="/adminplayer"
												>
													<div id="manage-players">
														Manage Players
													</div>
												</Link>
											</div>
											<div>
												<Link
													className="user-options-item"
													onClick={this.logout}
												>
													<div id="logout-admin">
														Logout
													</div>
												</Link>
											</div>
										</>
									) : (
										<div>
											<Link
												className="user-options-item"
												onClick={this.logout}
											>
												<div id="logout">Logout</div>
											</Link>
										</div>
									)
								) : null}
							</div>
						</li>
					</ul>
				</nav>
			</div>
		);
	}
}

const mapStateToProps = state => ({
	search: state.SearchReducer.search
});

export default connect(
	mapStateToProps,
	{ getSearch }
)(Navbar);
