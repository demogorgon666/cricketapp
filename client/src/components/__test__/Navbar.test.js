import React from "react";
import { shallow } from "enzyme";
import Navbar from "../common/Navbar";

const wrapper = shallow(<Navbar />);

describe("test for the text,css properties and icons on the navbar", () => {
  it("should have text brand CricketAlpha, matches, teams, players, stats", () => {
    // checks for no. of link present on navbar
    expect(wrapper.find(".link").length).toBe(6);

    // checks for the links text
    expect(
      wrapper
        .find(".link")
        .at(0)
        .text()
    ).toBe("Matches");
    expect(
      wrapper
        .find(".link")
        .at(1)
        .text()
    ).toBe("Teams");
    expect(
      wrapper
        .find(".link")
        .at(2)
        .text()
    ).toBe("Players");
    expect(
      wrapper
        .find(".link")
        .at(3)
        .text()
    ).toBe("Stats");

    // checks for the nav brand text
    expect(wrapper.find(".navBrand").text()).toBe("CricketAlpha");
  });
});
