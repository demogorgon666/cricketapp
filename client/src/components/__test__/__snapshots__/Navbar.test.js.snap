// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for the text, input, css properties and icons on the navbar and snapshot testing renders the component for snapshot testing 1`] = `
<div
  className="nav-parent"
>
  <nav
    className="menu"
  >
    <ul>
      <li
        className="nav-brand-li"
      >
        <Link
          id="nav-brand"
          onClick={[Function]}
          to="/"
        >
          CricketAlpha
        </Link>
      </li>
      <div
        className="genders"
      >
        <span
          className=""
          id="men"
          onClick={[Function]}
        >
          Men
        </span>
        <span
          className=""
          id="women"
          onClick={[Function]}
        >
          Women
        </span>
      </div>
      <input
        checked={true}
        className="menu-btn"
        id="menu-btn"
        onClick={[Function]}
        type="checkbox"
      />
      <label
        className="menu-icon"
        for="menu-btn"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <span
          className="bars"
        />
      </label>
      <li
        className="item"
        onClick={[Function]}
        style={
          Object {
            "display": "block",
          }
        }
      >
        <Link
          className="link"
          id="matches"
          to="/matches"
        >
          Matches
        </Link>
      </li>
      <li
        className="item"
        onClick={[Function]}
        style={
          Object {
            "display": "block",
          }
        }
      >
        <Link
          className="link"
          id="teams"
          to="/teams"
        >
          Teams
        </Link>
      </li>
      <li
        className="item"
        onClick={[Function]}
        style={
          Object {
            "display": "block",
          }
        }
      >
        <Link
          className="link"
          id="players"
          to="/players"
        >
          Players
        </Link>
      </li>
      <li
        className="search-box-li"
      >
        <input
          className="search-box"
          id="searchInput"
          name="searchInput"
          onChange={[Function]}
          placeholder="Search for Team or Player"
          value=""
        />
        <div
          className="search-result"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div>
            <Link
              className="search-list-item"
              to={
                Object {
                  "pathname": "/playerInfo/58",
                }
              }
            >
              <div
                style={
                  Object {
                    "backgroundImage": "url(data:image/jpeg;base64,undefined",
                  }
                }
              />
              <div
                className="search-suggestion"
                id="player-search-58"
              >
                JEC Franklin
              </div>
            </Link>
          </div>
          <div>
            <Link
              className="search-list-item"
              to={
                Object {
                  "pathname": "/teamdetails/10",
                }
              }
            >
              <div
                style={
                  Object {
                    "backgroundImage": "url(data:image/jpeg;base64,undefined",
                  }
                }
              />
              <div
                className="search-suggestion"
                id="team-search-10"
              >
                India
              </div>
            </Link>
          </div>
        </div>
      </li>
      <li
        className="login-icon"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <Link
          className="link"
          to="/login"
        >
          <i
            className="fas fa-user"
          />
        </Link>
        <div
          className="user-options-links"
        >
          <div>
            <Link
              className="user-options-item"
              to="/login"
            >
              <div
                id="login"
              >
                Login
              </div>
            </Link>
          </div>
          <div>
            <Link
              className="user-options-item"
              to="/register"
            >
              <div
                id="register"
              >
                Register
              </div>
            </Link>
          </div>
        </div>
      </li>
    </ul>
  </nav>
</div>
`;
